sermant-agentcore/sermant-agentcore-implement/src/test/java/io/sermant/implement/service/xds/utils/RdsProtocolTransformerTest.java

```diff
@@ -0,0 +1,223 @@
+ */
+
+package io.sermant.implement.service.xds.utils;
+
+import com.google.protobuf.BoolValue;
+import com.google.protobuf.UInt32Value;
+
+import io.envoyproxy.envoy.config.route.v3.HeaderMatcher;
+import io.envoyproxy.envoy.config.route.v3.Route;
+import io.envoyproxy.envoy.config.route.v3.Route.Builder;
+import io.envoyproxy.envoy.config.route.v3.RouteAction;
+import io.envoyproxy.envoy.config.route.v3.RouteConfiguration;
+import io.envoyproxy.envoy.config.route.v3.RouteMatch;
+import io.envoyproxy.envoy.config.route.v3.VirtualHost;
+import io.envoyproxy.envoy.config.route.v3.WeightedCluster;
+import io.envoyproxy.envoy.config.route.v3.WeightedCluster.ClusterWeight;
+import io.envoyproxy.envoy.type.matcher.v3.StringMatcher;
+import io.sermant.core.service.xds.entity.XdsRoute;
+import io.sermant.core.service.xds.entity.XdsRouteConfiguration;
+import io.sermant.core.service.xds.entity.XdsVirtualHost;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * @author daizhenyu
+ * @since 2024-08-24
+ **/
+public class RdsProtocolTransformerTest {
+    private int count = 1;
+
+    @Before
+    public void setUp() throws Exception {
+    }
+
+    @After
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void testGetRouteConfigurations() {
+        List<RouteConfiguration> routeConfigurations = Arrays.asList(
+                null,
+                createRouteConfiguration("exact", "exact"),
+                createRouteConfiguration("prefix", "prefix"),
+                createRouteConfiguration("prefix", "present")
+        );
+        List<XdsRouteConfiguration> result = RdsProtocolTransformer
+                .getRouteConfigurations(routeConfigurations);
+        result.sort(Comparator.comparing(XdsRouteConfiguration::getRouteConfigName));
+        Assert.assertEquals(3, result.size());
+
+        // pathType:exact headerType:exact
+        XdsRouteConfiguration routeConfiguration = result.get(0);
+        Assert.assertEquals("testRoute1", routeConfiguration.getRouteConfigName());
+        Map<String, XdsVirtualHost> virtualHosts = routeConfiguration.getVirtualHosts();
+        Assert.assertEquals(1, virtualHosts.size());
+        Assert.assertTrue(virtualHosts.containsKey("serviceA"));
+        XdsVirtualHost virtualHost = virtualHosts.get("serviceA");
+        List<XdsRoute> routes = virtualHost.getRoutes();
+        Assert.assertEquals(2, routes.size());
+        Set<Boolean> expectIsWeightedCluster = new HashSet<>();
+        expectIsWeightedCluster.add(true);
+        expectIsWeightedCluster.add(false);
+        Set<Boolean> actualIsWeightedCluster = new HashSet<>();
+        actualIsWeightedCluster.add(routes.get(0).getRouteAction().isWeighted());
```

Comment:
UT can be seperate to concrete test scenarios，it is hard to identify which functions are tested

Reply:
test method named getRouteConfigurations for different scenarios

Reply:
testGetRouteConfigurationswithscenarioA, testGetRouteConfigurationswithscenarioA is better